circle.yml

```diff
@@ -275,202 +311,211 @@ jobs:
-          key: v1-node6-clang39-debug
-          paths:
-            - node_modules
-            - /root/.ccache
-      - run:
-          name: Build node
-          command: make node
-      - run:
-          name: Run node tests
-          command: |
-            source scripts/circle_setup.sh
-            mapbox_install_logbt
-            mapbox_install_apitrace
-            mapbox_export_mesa_library_path
-            xvfb-run --server-args="-screen 0 1024x768x24" \
-              ./logbt -- apitrace trace --api=egl -v make test-node
-      - run:
-          name: Publish node package
-          when: on_success
-          command: platform/node/scripts/after_success.sh
-      - save_cache:
-          key: v1-node6-clang39-debug
-          paths:
-            - node_modules
-            - /root/.ccache
-      - store_artifacts:
-          path: mapbox-gl-js/test/integration/render-tests/index.html
-          destination: render-tests
+      - *generate-cache-key
+      - *restore-cache
+      - *reset-ccache-stats
+      - *build-node
+      - *show-ccache-stats
+      - *run-node-tests
+      - *publish-node-package
+      - *save-cache
+      - *upload-render-tests
 
 # ------------------------------------------------------------------------------
   linux-clang39-debug:
     docker:
-      - image: mbgl/ci:r3-linux-clang-3.9
+      - image: mbgl/ci:r4-linux-clang-3.9
     working_directory: /src
     environment:
       LIBSYSCONFCPUS: 6
       JOBS: 6
       BUILDTYPE: Debug
       WITH_EGL: 1
-      DISPLAY: :0
     steps:
       - checkout
-      - restore_cache:
-          key: v1-linux-clang39-debug
-          paths:
-            - node_modules
-            - /root/.ccache
-      - run:
-          name: Build linux
-          command: make linux
+      - *generate-cache-key
+      - *restore-cache
+      - *reset-ccache-stats
+      - *build-linux
+      - *build-benchmark
+      - *build-test
+      - *show-ccache-stats
+      - *run-unit-tests
+      - *save-cache
+
+# ------------------------------------------------------------------------------
+  linux-clang4-sanitize-address:
+    docker:
+      - image: mbgl/ci:r4-linux-clang-4
+    working_directory: /src
+    environment:
+      LIBSYSCONFCPUS: 6
+      JOBS: 6
+      BUILDTYPE: Sanitize
+      WITH_EGL: 1
+      GDB: '' # Do not run with GDB
+      CXXFLAGS: -fsanitize=address
+      LDFLAGS: -fsanitize=address
+      ASAN_OPTIONS: detect_leaks=0:color=always
+    steps:
+      - checkout
+      - *generate-cache-key
+      - *restore-cache
+      - *reset-ccache-stats
+      - *setup-llvm-symbolizer
+      - *build-test
+      - *show-ccache-stats
       - run:
-          name: Build benchmark
-          command: make benchmark
+          name: Run tests
+          command: |
+            xvfb-run --server-args="-screen 0 1024x768x24" make run-test 2> >(tee sanitizer 1>&2)
+            # Unfortunately, Google Test eats the status code, so we'll have to check the output.
```

Comment:
Google Test uses `fork` to execute the tests, and failing tests in those child processes probably return the correct status code. However, Google Test fails to interpret/forward that.